### PR TITLE
Adding case search frontend Part #2 Multiple appeals

### PR DESCRIPTION
**Pull request by tejans24**
connects #2342 

Use DEMO123 in the search for multiple appeals.

### CODE REVIEW

File: client/app/reader/CaseSelect.jsx [hunk 1]
```diff
@@ -82,7 +82,7 @@ class CaseSelect extends React.PureComponent {
       <div className="cf-app">
         <div className="cf-app-segment cf-app-segment--alt">
           <h1 className="welcome-header">Welcome to Reader!</h1>
-          <CaseSelectSearch history={this.props.history} />
+          <CaseSelectSearch history={this.props.history} feedbackUrl={this.props.feedbackUrl}/>
```
Comment (amprokop) [inline client/app/reader/CaseSelect.jsx]: Why not put this value in Redux state?

**Comment by amprokop** [comment]
General comment — when you open a PR that adds or changes the UI, can you get in the habit of putting a screenshot in the body of the PR? This helps designers keep track of changes and approve any deviations and it also helps your code reviewer with a frame of reference. I added a screenshot of your PR here.

**Comment by amprokop** [comment]
For this PR, I see that the spacing looks different from the mock— @abbyraskinUSDS might like to know this because we might need to have a follow up issue. What do you think of the spacing here, Abby?

**Comment by gnakm** [comment]
Can we add 15px between Veteran, Veteran ID, and Issues, that would be great

image

**Comment by tejans24** [comment]
Opps, I forgot to include the css related to this issue. I'll add that to this PR.

**Comment by tejans24** [comment]
And thanks @gnakm for the verification.

**Comment by tejans24** [comment]
@amprokop yea, it hasn't been a norm but it does make sense. We should probably start doing that for all of our PRs.

File: client/app/reader/CaseSelect.jsx (hunk 1 above)
Comment (tejans24) [inline client/app/reader/CaseSelect.jsx]: It seemed excessive since it's only used in one place in Reader. I can do it if we think it's worth.

**Comment by tejans24** [comment]
screen shot 2017-08-23 at 3 55 22 pm

File: client/app/reader/actions.js [hunk 2]
```diff
@@ -627,6 +627,11 @@ export const fetchAppealUsingVeteranIdFailed = () => ({
   type: Constants.RECEIVE_APPEALS_USING_VETERAN_ID_FAILED
 });
 
+export const onReceiveMultipleAppealsWithVeteranId = (appeals) => ({
+  type: Constants.RECEIVE_MULTIPLE_APPEALS_USING_VETERAN_ID,
```
Comment (amprokop) [inline client/app/reader/actions.js]: Do we *need* a new action for this, or does it make our state unnecessarily complex? I'm wondering why we don't just always store appeals in the same array, then just store `selectedAppealId` as a separate string.

Comment (amprokop) [inline client/app/reader/actions.js]: Did you consider that option, and were there downsides?

File: client/app/reader/actions.js [hunk 3]
```diff
@@ -642,6 +647,8 @@ export const fetchAppealUsingVeteranId = (veteranId) => (
       // automatically select that appeal for viewing
       if (numOfAppeals === 1) {
         dispatch(caseSelectAppeal(returnedObject.appeals[0]));
+      } else if (numOfAppeals > 1) {
```
Comment (amprokop) [inline client/app/reader/actions.js]: Similar to the top — I'm wondering if we're making the state & code unneccessarily complex and hard to understand.

**Comment by amprokop** [comment]
Btw I'm down to help out with some of the refactoring issues I'm insisting on, don't worry :)

File: client/app/reader/CaseSelectSearch.jsx [hunk 4]
```diff
@@ -30,13 +33,50 @@ class CaseSelectSearch extends React.PureComponent {
     }
   };
 
+  handleModalClose = () => {
+    // clearing the state of the modal
+    this.setState({ selectedAppealVacolsId: null });
```
Comment (tejans24) [inline client/app/reader/CaseSelectSearch.jsx]: yea, I like keeping things in the state if they are coupled to the component.

1). No other component in the app needs to access this variable from the store
2). It's tightly coupled to this component
3). It's basically a simple modal toggle
4). The result from the modal is only used in this component.

It made sense to keep it in the react component state than inserting it in Redux.

File: client/app/reader/actions.js (hunk 2 above)
Comment (tejans24) [inline client/app/reader/actions.js]: I did consider that and I thought the complexity was similar for both cases. It's a matter of where it's handled.

-  Fetch all appeals for the veteran id 
- When a veteran id only has one appeal -> We can select this appeal.  Redirect the user to the document list view with that selected appeal and it's documents in the Component.
- When a Veteran id has multiple appeals -> Store those appeals in an array in the redux store, show the array options to the user and let them select an appeal they would like to view. Then redirect the user to the document list view with that appeal in the component.

In the 2nd case: 
* Fetch all appeals for the veteran id and store them in an array in the redux store.
* Check if the array only has one element in the component, redirect the user to the document list page with the selected appeal.
* If the array contains multiple elements, show the array options to the user and let them select an appeal they would like to view. Then redirect the user to the document list view with that appeal in the component.

I thought having the logic in the action makes it simple to understand than if it's in the component. If you think otherwise, I can make the change. I'll create an issue for this either way.

Comment (tejans24) [inline client/app/reader/actions.js]: https://github.com/department-of-veterans-affairs/caseflow/issues/3059

File: client/app/reader/actions.js [hunk 5]
```diff
@@ -627,6 +627,11 @@ export const fetchAppealUsingVeteranIdFailed = () => ({
   type: Constants.RECEIVE_APPEALS_USING_VETERAN_ID_FAILED
 });
 
+export const onReceiveMultipleAppealsWithVeteranId = (appeals) => ({
```
Comment (nickheiner-usds) [inline client/app/reader/actions.js]: Please always add a `meta.analytics` field when adding a new action. You can follow the pattern from the existing actions.

File: client/app/reader/CaseSelectSearch.jsx [hunk 6]
```diff
@@ -47,6 +87,36 @@ class CaseSelectSearch extends React.PureComponent {
         onSubmit={this.searchOnChange}
         submitUsingEnterKey
       />
+      { _.size(caseSelect.receivedAppeals) ? <Modal
```
Comment (nickheiner-usds) [inline client/app/reader/CaseSelectSearch.jsx]: No need for the ternary here. Instead of:

```jsx
{ _.size(caseSelecct.receivedAppeals) ? <Modal /> : '' }
```

we can just do:

```jsx
{ Boolean(_.size(caseSelecct.receivedAppeals)) && <Modal /> }
```

File: client/app/reader/CaseSelectSearch.jsx [hunk 7]
```diff
@@ -30,13 +33,50 @@ class CaseSelectSearch extends React.PureComponent {
...
     if (_.size(text)) {
       this.props.fetchAppealUsingVeteranId(text);
     }
   }
 
   render() {
+    const { caseSelect } = this.props;
+
+    const createAppealOptions = (appeals) => {
+      return appeals.map((appeal) => {
```
Comment (nickheiner-usds) [inline client/app/reader/CaseSelectSearch.jsx]: When a function is only a single return value, please use the shorter syntax. This should be:

```js
    const createAppealOptions = (appeals) =>
       appeals.map((appeal) => {
```

File: client/app/reader/CaseSelectSearch.jsx (hunk 4 above)
Comment (nickheiner-usds) [inline client/app/reader/CaseSelectSearch.jsx]: In the past, our pattern on Reader has been to put everything in Redux. I agree that there are reasonable differences of opinion on how much state should go in Redux, and state that is only used in one place may not need to go there. I'd defer to @mdbenjam on what direction he wants Reader to take.

I would point out that if you use Redux, you get analytics integration for free. With this code, we are not getting GA tracking.

I would also point out that the more Redux we use, the easier it is to test our entire app via unit tests with the reducer, which are easier to understand and maintain than tests that involve rendering components.

And, being 100% redux enables more hot reloading and time travel debugging, which I find makes development nicer.

**tl;dr** Reader has been following a convention of putting everything into Redux. There could be reasons to deviate from that, but we should do so in a thoughtful way.

Comment (mdbenjam) [inline client/app/reader/CaseSelectSearch.jsx]: I agree with what Nick said. AFAIK all new state we've added since around May has been added to Redux. Any remaining local state is left over from before that time and I'd like to move it to Redux for consistency.

Comment (tejans24) [inline client/app/reader/CaseSelectSearch.jsx]: That's a good point about the free analytics integration that I did not consider. It makes sense to move this into the redux store.
